kroxylicious-sample/src/test/java/io/kroxylicious/sample/SampleFilterIntegrationTest.java

```diff
@@ -0,0 +1,154 @@
+        tester = kroxyliciousTester(withDefaultFilters(proxy(cluster))
+                .addToFilters(new FilterDefinitionBuilder(SampleContributor.SAMPLE_PRODUCE).withConfig(PRODUCE_CONFIG).build())
+                .addToFilters(new FilterDefinitionBuilder(SampleContributor.SAMPLE_FETCH).withConfig(FETCH_CONFIG).build()));
+    }
+
+    @AfterEach
+    public void afterEach() {
+        tester.close();
+    }
+
+    @Test
+    public void sampleFilterWillTransformRoundTripTest(Admin admin) {
+        try (Producer<String, String> producer = tester.producer();
+                Consumer<String, byte[]> kafkaClusterConsumer = new KafkaConsumer<>(mergeMaps(cluster.getKafkaClientConfiguration(),
+                        Map.of(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class, ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
+                                ByteArrayDeserializer.class, ConsumerConfig.GROUP_ID_CONFIG, "group-id-0", ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest")));
+                Consumer<String, byte[]> proxyConsumer = tester.consumer(Serdes.String(), Serdes.ByteArray(),
+                        Map.of(ConsumerConfig.GROUP_ID_CONFIG, "group-id-1", ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest"))) {
+            admin.createTopics(List.of(new NewTopic(TOPIC_NAME, TOPIC_PARTITIONS, TOPIC_REPLICATION))).all().get(TIMEOUT_SECONDS, TimeUnit.SECONDS);
+            producer.send(new ProducerRecord<>(TOPIC_NAME, PRE_TRANSFORM_VALUE)).get(TIMEOUT_SECONDS, TimeUnit.SECONDS);
+            // get from cluster
+            kafkaClusterConsumer.subscribe(List.of(TOPIC_NAME));
+            ConsumerRecords<String, byte[]> kafkaClusterPoll = kafkaClusterConsumer.poll(Duration.ofSeconds(TIMEOUT_SECONDS));
+            if (kafkaClusterPoll.count() != 1) {
+                fail(String.format("Sent 1 record but received %d records from Kafka Cluster Consumer.", kafkaClusterPoll.count()));
+            }
+            ConsumerRecord<String, byte[]> clusterRecord = kafkaClusterPoll.records(TOPIC_NAME).iterator().next();
+            // check cluster record value is correct
+            assertEquals(PRODUCE_TRANSFORM_VALUE, new String(clusterRecord.value(), StandardCharsets.UTF_8));
+            // get from proxy
+            proxyConsumer.subscribe(List.of(TOPIC_NAME));
+            ConsumerRecords<String, byte[]> proxyPoll = proxyConsumer.poll(Duration.ofSeconds(TIMEOUT_SECONDS));
+            if (proxyPoll.count() != 1) {
+                fail(String.format("Sent 1 record but received %d records from Proxy Consumer.", proxyPoll.count()));
+            }
+            ConsumerRecord<String, byte[]> proxyRecord = proxyPoll.records(TOPIC_NAME).iterator().next();
+            // check proxy record is correct
+            assertEquals(FETCH_TRANSFORM_VALUE, new String(proxyRecord.value(), StandardCharsets.UTF_8));
+        }
+        catch (Exception e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
Is this block needed? I think we are just using the try block for the auto close.

Reply:
It's there for the future get exceptions.